hummingbot/strategy/cross_exchange_market_making/cross_exchange_market_making.pyx

```diff
@@ -847,6 +847,9 @@ cdef class CrossExchangeMarketMakingStrategy(StrategyBase):
 
             taker_slippage_adjustment_factor = Decimal("1") + self._slippage_buffer
             taker_balance = taker_balance_in_quote / (taker_price * taker_slippage_adjustment_factor)
+            if Decimal.is_nan(taker_price):
+                taker_balance = maker_balance
```

Comment:
Hi, sorry I don't think it is a good idea to set taker_balance = maker_balance here just because of too much quote balance. 
For example, if base contains 1k token while quote contain 100 token (converted from quote amount) and the order book depth only has 90 tokens. this would still result in a order size of 1k token even though the taker could only get 100 token. This order would then immediately get canceled when c_check_if_sufficient_balance is called.

This issue is mainly due to the taker_market order depth does not have sufficient depth due to large balance which we can limit by using the user_order instead. we can also change to using an existing function here on line 841.

`self.c_calculate_effective_hedging_price(market_pair, is_bid, user_order)`